Dockerfile

```diff
@@ -31,6 +31,14 @@ RUN apt update -y -q && apt upgrade -y -q && apt update -y -q && \
     lldb \
     gettext \
     ninja-build \
+    file \
+    perl \
+    xxd \
+    mesa-common-dev \
```

Comment:
so long as the runtime for these are not required to run the compilers?

Reply:
I'm pretty sure they're just build dependencies. mesa-common-dev definitely is required after building.

perl and file are needed for hipcc, but I was going to just set up CE to use the ROCm clang directly, so they're not needed.

I'm pretty sure xxd is also just a build dependency. It was just needed to compile rocr.

Reply:
I just did a couple more tests. perl is part of the ubuntu:20.04 base docker image, so I didn't try removing it, but my local copy of CE works fine without mesa, xdd or file. 

I can confirm that they're only needed at library build time. They are not needed for compiling HIP code.

Reply:
Fantastic! I'll try and get this all reviewed and tested tongith, but if not, Friday is my next concerted Compiler Explorer time.